src/app/app_main.c

```diff
@@ -0,0 +1,6 @@
+#include <stdint.h>
+#include "app_header.h"
+
+int main() {
```

Comment:
```suggestion
int main(void) {
```
Check out section 6.7.7.4, paragraph 9 and 13 of the C23 draft: https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3220.pdf

tldr: the way you have `main` defined means that it takes an unspecified amount of parameters (before C23), *however*, with how you call it from `_start`, it takes no arguments.

Reply:
FYI: an unspecified parameter in C23 is the equivalent of adding `void`, but small detail to be aware of in older versions.